app/transport/rest/user_test.go

```diff
@@ -0,0 +1,131 @@
+		expResponseBody    string
+	}{
+		"succes": {
+			requestBody: `{"id": "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc"}`,
+			usecaseConstructor: func(ctrl *gomock.Controller) UserUsecase {
+				mock := NewMockUserUsecase(ctrl)
+
+				mock.EXPECT().Delete(entities.User{
+					ID: "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc",
+				}).Return(nil).Times(1)
+
+				return mock
+			},
+			expStatusCode: 204,
+		},
+		"invalid_undefined_id": {
+			requestBody: `{"id": "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc"}`,
+			usecaseConstructor: func(ctrl *gomock.Controller) UserUsecase {
+				mock := NewMockUserUsecase(ctrl)
+				mock.EXPECT().Delete(entities.User{
+					ID: "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc",
+				}).Return(globals.ErrNotFound).Times(1)
+
+				return mock
+			},
+			expResponseBody: `
+									{
+										"message": "Not found",
+										"details": "no user found in DB"
+									}`,
+			expStatusCode: 404,
+		},
+		"invalid_bad_id_too_long": {
+			requestBody: `{"id": "1acef78b-55ef-43b1-8c2b-c9e36c9c11111"}`,
+			usecaseConstructor: func(ctrl *gomock.Controller) UserUsecase {
+				return nil
+			},
+			expResponseBody: `
+									{
+			  						"message": "Bad request",
+			 						"details": "invalid uuid: invalid UUID length: 37"
+									}`,
+			expStatusCode: 400,
+		},
+		"invalid_bad_id_too_short": {
+			requestBody: `{"id": "1acef78b-55ef-43b1-8c2b-c9e36c9c280"}`,
+			usecaseConstructor: func(ctrl *gomock.Controller) UserUsecase {
+				return nil
+			},
+			expResponseBody: `
+									{
+			  						"message": "Bad request",
+								    "details": "invalid uuid: invalid UUID length: 35"
+									}`,
+			expStatusCode: 400,
+		},
+		"invalid_bad_id_bad_symbpls": {
+			requestBody: `{"id": "%acef78b-55ef-43b1-8c2b-c9e36c9c2807"}`,
+			usecaseConstructor: func(ctrl *gomock.Controller) UserUsecase {
+				return nil
+			},
+			expResponseBody: `
+									{
+			  						"message": "Bad request",
+								    "details": "invalid uuid: invalid UUID format"
+									}`,
+			expStatusCode: 400,
+		},
+		"invalid_internal_server_error": {
+			requestBody: `{"id": "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc"}`,
+			usecaseConstructor: func(ctrl *gomock.Controller) UserUsecase {
+				mock := NewMockUserUsecase(ctrl)
+				mock.EXPECT().Delete(entities.User{
+					ID: "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc",
+				}).Return(errors.New("Internal error")).Times(1)
+
+				return mock
+			},
+			expResponseBody: `
+									{
+										"message": "Internal server error",
+										"details": ""
+									}`,
+			expStatusCode: 500,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			uh := UserHandler{
+				usecase: tt.usecaseConstructor(ctrl),
+				logger:  logger,
+			}
+
+			resp := httptest.NewRecorder()
+			uh.DeleteUser(resp, httptest.NewRequest(http.MethodDelete, "*", strings.NewReader(tt.requestBody)))
+			require.Equal(t, tt.expStatusCode, resp.Code)
+			if name != "succes" {
```

Comment:
That's not good.
We should not have specific conditions here.
If you don't need to check body in this test case - try to describe that in a test case itself.